src/BlockTypesController.php

```diff
@@ -141,14 +145,15 @@ public function block_hydration_wrapper( $block_content, $block, $instance ) {
 			$block['blockName'],
 			array(
 				'woocommerce/all-products',
+				'woocommerce/cart-order-summary-coupon-form-block',
 			),
 			true
 		) ) {
 			return $block_content;
 		}
 
 		$block_type       = $instance->block_type;
-		$attr_definitions = $block_type->attributes;
+		$attr_definitions = isset( $block_type ) ? $block_type->attributes : [];
```

Comment:
Hey Carlos, why are those `isset` needed? Doesn't the Summary Coupon Form block have a `$block_type` defined?

Reply:
@luisherranz, @c4rl0sbr4v0, I think inner blocks' types are not registered in PHP, only on the editor. 🤔

Maybe we should register them first to make it work as intended!

Reply:
Yes, it'd be better to make server registration a requirement.

Reply:
I made a test registering-deregistering and is for that reason. If the block is not registered in PHP, `$block_type` is null.

Adding a notice may be too much 😅 , what could be an option to tell the developer this requirement?
![Screenshot 2022-08-10 at 13 23 51](https://user-images.githubusercontent.com/37012961/183889697-065da61b-e1f4-4766-b8f8-da583e26fec0.jpg)

Reply:
I'm not sure yet how to best present this to the developer, but it seems there's a number of requirements for our BHE technique to work; another one (that I discovered while working on #21) was that the block needed to have a `block.json` (rather than doing the client-side registration in JS).

(The reason is similar to what we're dealing here: Our BHE code needs to know the block's attribute definitions on the server side. A `block.json` provides that, while client-side JS of course doesn't.)

Reply:
I think it's fine to add these missing requirements on an ad-hoc basis for now (like you're doing in 28c9f38), and maybe start a list to keep track of them. At this point, it's just:

- Needs a `block.json`.
- Needs to be registered on the server-side.

Reply:
> * Needs a `block.json`.
> * Needs to be registered on the server-side.

Let's put this on BHE repo.